Updated dead link in "how-to-spend-company-money"

fmartingr opened the pull request:
#### Summary

Update the recommended books/resources for training link[^1] from the How to spend company money page since it does not exist.

[^1]: https://handbook.mattermost.com/operations/research-and-development/training

fmartingr committed:
chore: updated link to training resources

mattermod commented:
Hello @fmartingr,

Thanks for your pull request! A Core Committer will review your pull request soon. For code contributions, you can learn more about the review process [here](https://developers.mattermost.com/contribute/getting-started/code-review/).

justinegeffen commented:
Thank you! This link has just been updated so I've added the new link as a suggestion. Thanks for finding this page, I didn't realize it was out of date. :)

justinegeffen commented on operations/finance/staff-member-expenses/how-to-spend-company-money.md:
```diff
@@ -41,7 +41,6 @@ While we're sharing average costs for various items below, you should use your b
 * Laptop stand: Average price **180 USD**
 * Laptop carrying bag: Average price **60 USD**
 * Work-related books: **30 USD** per month or **360 USD** annually
-  * [Recommended resources](https://handbook.mattermost.com/operations/research-and-development/training)
```
https://handbook.mattermost.com/operations/research-and-development/product/product-management-team-handbook/professional-development#additional-training-resources

fmartingr commented:
> Thank you! This link has just been updated so I've added the new link as a suggestion. Thanks for finding this page, I didn't realize it was out of date. :)

I have updated the link with the correct one :)

justinegeffen reviewed (approved):
Thank you so much! :)
